libs/datamodel/diagnostics/src/warning.rs

```diff
@@ -2,7 +2,7 @@ use crate::Span;
 
 /// A non-fatal warning emitted by the schema parser.
 /// For fancy printing, please use the `pretty_print_error` function.
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, PartialEq, Eq, Clone)]
```

Comment:
For convenience for get_config. We want the warnings in both the Diagnostics and the Configuration structs, so we clone. It could definitely use some improvement. I think we should deprecate `Configuration` as a standalone thing, it can live inside `ValidatedSchema` always (and then we have two functions producing a `ValidatedSchema`, with the config-only one returning the same type, but doing less work, since it only cares about configuration). I could do that in this PR if that feels better, but the diff would get larger.

Reply:
No don't do it here :D Save it for later. I was just wondering the cloning.